geopandas/io/sql.py

```diff
@@ -12,35 +12,50 @@ def read_postgis(sql, con, geom_col='geom', crs=None, index_col=None,
     Returns a GeoDataFrame corresponding to the result of the query
     string, which must contain a geometry column.
 
-    Examples:
-    sql = "SELECT geom, kind FROM polygons;"
-    df = geopandas.read_postgis(sql, con)
-
     Parameters
     ----------
-    sql: string
-    con: DB connection object or SQLAlchemy engine
-    geom_col: string, default 'geom'
+    sql : string
+        SQL query to execute in selecting entries from database.
+    con : DB connection object or SQLAlchemy engine
+        Active connection to the database to query.
+    geom_col : string, default 'geom'
         column name to convert to shapely geometries
-    crs: optional
+    crs : dict or str, optional
         CRS to use for the returned GeoDataFrame
 
     See the documentation for pandas.read_sql for further explanation
     of the following parameters:
     index_col, coerce_float, params
 
+    Returns
+    -------
+    geodataframe : GeoDataFrame
+
+    Example
+    -------
+    >>> sql = "SELECT geom, kind FROM polygons;"
+    >>> df = geopandas.read_postgis(sql, con)
     """
-    df = read_sql(sql, con, index_col=index_col, coerce_float=coerce_float,
-                  params=params)
 
-    if geom_col not in df:
-        raise ValueError("Query missing geometry column '{0}'".format(
-            geom_col))
+    # add SRID call to sql str
+    sql_srid = sql.lower().replace(" from ",
+                                   ",st_srid({}) from ".format(geom_col))
```

Comment:
This feels a bit brittle. Not necessarily every query has the 'from' format (you can run a procedure I think? long time ago I actively used sql)

Reply:
I can certainly believe that. When I a) didn't see any way to get the SRID from the individual entry and b) didn't see a good way to run a second query on the table without changing the API, I thought I'd throw this out. Since you marked this as "good first issue", I was hoping you'd have something much better in mind that you'd share once I threw this up. :wink: